File: backend/app/schemas/discord_voice.py
from pydantic import BaseModel, Field


class DiscordVoiceUserOut(BaseModel):
    user_id: str
    nickname: str


class DiscordVoiceChannelOut(BaseModel):
    channel_id: str
    name: str
    users: list[DiscordVoiceUserOut] = Field(default_factory=list)


class DiscordVoiceStatusOut(BaseModel):
    users: list[DiscordVoiceUserOut] = Field(default_factory=list)
    channels: list[DiscordVoiceChannelOut] = Field(default_factory=list)
    user_count: int = 0
    guild_name: str = ""
    configured: bool = False

File: frontend/src/types/discord-voice.ts
export interface DiscordVoiceUser {
  user_id: string
  nickname: string
}

export interface DiscordVoiceChannel {
  channel_id: string
  name: string
  users: DiscordVoiceUser[]
}

export interface DiscordVoiceStatus {
  users: DiscordVoiceUser[]
  channels: DiscordVoiceChannel[]
  user_count: number
  guild_name: string
  configured: boolean
}
